betty/extension/npm/__init__.py

```diff
@@ -78,128 +70,14 @@ def details(self) -> Str:
-
-    @property
-    def _extension_types_with_built_assets(self) -> set[type[_NpmBuilder & Extension]]:
-        return {
-            extension_type
-            for extension_type
-            in self._extension_types
-            if is_assets_build_directory_path(_get_assets_build_directory_path(extension_type))
-        }
-
-    def is_met(self) -> bool:
-        return self._extension_types <= self._extension_types_with_built_assets
-
-    def summary(self) -> Str:
-        return self._summary
-
-    def details(self) -> Str:
-        return self._details
-
-
-@unique
-class _NpmBuilderCacheScope(IntFlag):
-    BETTY = auto()
-    PROJECT = auto()
-
-
-class _NpmBuilder:
-    async def npm_build(self, working_directory_path: Path, assets_directory_path: Path) -> None:
-        raise NotImplementedError(repr(self))
-
-    @classmethod
-    def npm_cache_scope(cls) -> _NpmBuilderCacheScope:
-        return _NpmBuilderCacheScope.PROJECT
-
-
-def discover_npm_builders() -> set[type[_NpmBuilder & Extension]]:
-    """
-    Gather all extensions that are npm builders.
-    """
-    return {
-        extension_type
-        for extension_type
-        in discover_extension_types()
-        if issubclass(extension_type, _NpmBuilder)
-    }
-
-
-def _get_assets_directory_path(extension_type: type[_NpmBuilder & Extension]) -> Path:
-    assert issubclass(extension_type, Extension)
-    assert issubclass(extension_type, _NpmBuilder)
-    assets_directory_path = extension_type.assets_directory_path()
-    if not assets_directory_path:
-        raise RuntimeError(f'Extension {extension_type} does not have an assets directory.')
-    return assets_directory_path / _Npm.name()
-
-
-def _get_assets_src_directory_path(extension_type: type[_NpmBuilder & Extension]) -> Path:
-    return _get_assets_directory_path(extension_type) / 'src'
-
-
-def _get_assets_build_directory_path(extension_type: type[_NpmBuilder & Extension]) -> Path:
-    return _get_assets_directory_path(extension_type) / 'build'
-
-
-async def build_assets(extension: _NpmBuilder & Extension) -> Path:
-    """
-    Build the npm assets for an extension.
-    """
-    assets_directory_path = _get_assets_build_directory_path(type(extension))
-    await _build_assets_to_directory_path(extension, assets_directory_path)
-    return assets_directory_path
-
-
-async def _build_assets_to_directory_path(extension: _NpmBuilder & Extension, assets_directory_path: Path) -> None:
-    assert isinstance(extension, Extension)
-    assert isinstance(extension, _NpmBuilder)
-    with suppress(FileNotFoundError):
-        await asyncio.to_thread(shutil.rmtree, assets_directory_path)
-    os.makedirs(assets_directory_path)
-    async with TemporaryDirectory() as working_directory_path_str:
-        working_directory_path = Path(working_directory_path_str)
-        await extension.npm_build(Path(working_directory_path), assets_directory_path)
-
-
 class _Npm(Extension):
```

Comment:
Replace references to module's old _Nom build directories. There may be some in our shell scripts.